lms/djangoapps/courseware/models.py

```diff
@@ -29,10 +30,41 @@
 log = logging.getLogger("edx.courseware")
 
 
+def chunks(items, chunk_size):
+    """
+    Yields the values from items in chunks of size chunk_size
+    """
+    items = list(items)
+    return (items[i:i + chunk_size] for i in xrange(0, len(items), chunk_size))
+
+
+class ChunkingManager(models.Manager):
+    """
+    :class:`~Manager` that adds an additional method :meth:`chunked_filter` to provide
+    the ability to make select queries with specific chunk sizes.
+    """
+    def chunked_filter(self, chunk_field, items, **kwargs):
+        """
+        Queries model_class with `chunk_field` set to chunks of size `chunk_size`,
+        and all other parameters from `**kwargs`.
```

Comment:
This could be clearer.  it doesn't mention `items`, which is the sequence of values chunked, and applied to `chunk_field`.
If this is only a limitation on sqlite3, should we try to avoid the chunking when querying MySQL? To avoid multiple round trips?